src/Appwrite/Platform/Modules/Compute/Base.php

```diff
@@ -0,0 +1,191 @@
+<?php
+
+namespace Appwrite\Platform\Modules\Compute;
+
+use Appwrite\Event\Build;
+use Appwrite\Extend\Exception;
+use Utopia\CLI\Console;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Helpers\ID;
+use Utopia\Database\Helpers\Permission;
+use Utopia\Database\Helpers\Role;
+use Utopia\Database\Validator\Authorization;
+use Utopia\Platform\Action;
+use Utopia\Swoole\Request;
+use Utopia\System\System;
+use Utopia\VCS\Adapter\Git\GitHub;
+use Utopia\VCS\Exception\RepositoryNotFound;
+
+class Base extends Action
+{
+    public function redeployVcsFunction(Request $request, Document $function, Document $project, Document $installation, Database $dbForProject, Build $queueForBuilds, Document $template, GitHub $github)
```

Comment:
By all logic, this shouldn't be:

1. A functions in a controller
2. Used as base Action for *all* other actions
3. A public method

True to both `redeployVcsFunction ` and `redeployVcsSite` also this defeats the all purpose of the abstraction, both methods here don't make sense at all.

Reply:
Agree, same here: https://github.com/appwrite/appwrite/blob/9893f77e6dcc268fbb73dea57d3fa1bfc6e0c94f/app/controllers/api/vcs.php#L63

We also never addressed the fact that site deployment creation endpoint and function deployment creation are 99% same. They should also be much much simpler.

I dont think we have time to focus on this for 1.7 release. I vote to improve this later. Having structure that allows us to fix it is good first step - better than what we had before.